packages/types/src/codec/utils/encodeTypes.ts

```diff
@@ -18,25 +26,25 @@ export function encodeEnum (sub: TypeDef[]): string {
     }]`;
   }
 
-  return `{${
+  return `{ "_enum": { ${
     sub
-      .map(({ name, type }: TypeDef): string => `"${name}": ${type}`)
+      .map((type: TypeDef): string => `"${type.name}": "${encodeTypeWithParams(type)}"`)
       .join(', ')
-  }}`;
+  }} }`;
 }
 
 export function encodeStruct (sub: TypeDef[]): string {
-  return `{${
+  return `{ "_enum": {${
```

Comment:
This can't be right or am I missing something, i.e. struct !== enum

Reply:
You're right, misunderstood your decodeStruct